orion/primitives/azure_anomaly_detector.py

```diff
@@ -0,0 +1,175 @@
+import logging
+from datetime import datetime
+
+import numpy as np
+from azure.cognitiveservices.anomalydetector import AnomalyDetectorClient
+from azure.cognitiveservices.anomalydetector.models import APIErrorException, Point, Request
+from msrest.authentication import CognitiveServicesCredentials
+
+LOGGER = logging.getLogger(__name__)
+
+
+def split_sequence(X, index, target_column, sequence_size, pad_size):
+    """Split sequences of time series data.
+
+    The function creates a list of input sequences by splitting the input sequence
+    into partitions with a specified size and pad it be a spceificied size.
+
+    Args:
+        X (ndarray):
+            N-dimensional value sequence to iterate over.
+        index (ndarray):
+            N-dimensional index sequence to iterate over.
+        target_column (int):
+            Indicating which column of X is the target.
+        sequence_size (int):
+            Length of the input sequences.
+        pad_size (int):
+            Length of the values from previous window.
+
+    Returns:
+        list:
```

Comment:
I think the type should be `tuple` instead of `list` (or, actually, `tuple[np.ndarray]`).